AzureMonitorAgent/agent.py

```diff
@@ -1181,11 +1181,12 @@ def metrics_watcher(hutil_error, hutil_log):
                 f.close()
             
             if fluent_port != '' and os.path.isfile(FluentCfgPath):
-                portSetting = "    Port    "  + fluent_port
-                defaultPortSetting = 'Port    '
+                portSetting = "    Port                       "  + fluent_port + "\n"
+                defaultPortSetting = 'Port'
                 portUpdated = False
                 with open(FluentCfgPath) as f:
-                    if portSetting not in f.read():
+                    found = re.search(".*Port.*" + fluent_port, f.read())
```

Comment:
Can we tighten up the regex to use a start of line anchor and only match "Port" surrounded by whitespace? Just thinking about cases where "Port" is elsewhere in the config file (maybe as part of a custom log file path).